src/as-cache.c

```diff
@@ -1995,3 +1995,16 @@ as_cache_search (AsCache *cache, const gchar * const *terms, gboolean sort, GErr
 
 	return g_steal_pointer (&results);
 }
+
+gboolean
+as_cache_get_is_empty(AsCache* cache)
+{
+	g_autoptr(GError) error = NULL;
+	g_autoptr(GPtrArray) components = as_cache_query_components (cache,
```

Comment:
This is *extremely* inefficient, but let's ignore that for now, I can alter the code later to make it more efficient.

Reply:
Yes, that's what I thought. I imagine it will need changing libxmlb.